uaclient/tests/test_contract.py

```diff
@@ -96,6 +97,32 @@ def test_overrides_applied_before_comparison(
 
 @mock.patch(M_PATH + "UAContractClient")
 class TestGetAvailableResources:
+    def test_request_resources_error_on_network_disconnected(self, client):
```

Comment:
If I understand correctly, this test is just checking that `get_available_resources` doesn't catch `util.UrlError`? This seems like a lot of test code to ensure that, would it be simpler if instead of raising it specifically for this URL we just raised for any URLs? It feels like something like this could potentially replace diff lines 104-121:

```py
client.return_value.request_url.side_effect = util.UrlError(...)
```

Reply:
great suggestion thanks for thinking that through. I've patched the UAContractClient object's request_url method.